draft-yorgos-dnsop-dry-run-dnssec.md

```diff
@@ -0,0 +1,302 @@
+ street = "Science Park 400"
+ city = "Amsterdam"
+ code = "1098 XH"
+ country = "Netherlands"
+
+[[author]]
+initials="W."
+surname="Toorop"
+fullname="Willem Toorop"
+organization = "NLnet Labs"
+[author.address]
+ email = "willem@nlnetlabs.nl"
+[author.address.postal]
+ street = "Science Park 400"
+ city = "Amsterdam"
+ code = "1098 XH"
+ country = "Netherlands"
+%%%
+
+
+.# Abstract
+
+This document describes a method called "dry-run DNSSEC" that allows for
+testing DNSSEC deployments without affecting the DNS service in case of
+configuration errors. It accomplishes that by introducing a new DS Type Digest
+Algorithm that signals to validating resolvers that dry-run DNSSEC is used for
+the zone. DNSSEC errors are then reported with DNS Error Reporting, but the
+bogus answer is withheld and instead resolvers fallback from dry-run DNSSEC and
+provide the answer they would normally produce for the zone. Further options
+are presented for testing key rollovers with this method and an option for
+clients to opt-in for dry-run DNSSEC errors and allow for end-to-end DNSSEC
+testing.
+
+
+{mainmatter}
+
+
+# Introduction
+
+DNSSEC was introduced to provide DNS with data origin authentication and data
+integrity. This introduced quite an amount of complexity and fragility to the
+DNS which in turn still hinders general adoption. When an operator decides to
+publish a newly signed zone there is no way to realistically check that DNS
+will not break for the zone.
+
+This document describes a method called "dry-run DNSSEC" that gives confidence
+to operators to adopt DNSSEC by introducing a new special DS Type Digest
+Algorithm. Resolvers that don't support the algorithm continue to treat the
+delegation as insecure [@RFC6840, see, section 5.2]. Validating resolvers are
+signaled to treat the delegation as being in an intermediate test step for
+DNSSEC. Valid answers yield authentic data (AD) responses. Clients that expect
+the AD flag can already profit from the transition. Invalid answers instead of
+SERVFAIL yield the insecure data. This is of course not proper data integrity
+but the delegation should not be considered DNSSEC signed at this point.
+Together with DNS Error Reporting support, which is essential for dry-run
+DNSSEC, DNSSEC health is reported back to the operator.
+
+The signed zone is publicly deployed but DNSSEC configuration errors cannot
+break DNS resolution yet. DNSSEC health feedback can pinpoint potential issues
+back to the operator. When the operator is confident that the DNSSEC adoption
+does not introduce DNS breakage, the real DS record can be published on the
+parent zone and that concludes the actual DNSSEC deployment.
+
+Dry-run DNSSEC can further be used on already singed zones to test key
+rollovers. In this case a dry-run DS record for the future key is used next to
+the current DS record which itself needs to be also presented in the dry-run
+format. Validating resolvers that understand dry-run DNSSEC first try to
```

Comment:
Why does the current DS needs to be present as dry-run as well?

```suggestion
the current DS record. Validating resolvers that understand dry-run DNSSEC first try to
```

Reply:
To simulate the resolver's DS picking behavior.